functorch/_src/aot_autograd.py

```diff
@@ -167,86 +181,336 @@ def posthook(grad_input, grad_output):
 #
 # TODO: Provide a faster version of this that assumes flat arguments
 # (so no pytree necessary)
-def detach_and_functionalize_pure(f, preserve_requires_grad=True):
+def run_functionalized_fw_and_collect_metadata(f):
```

Comment:
sure would be useful to know what the return type of this function is

Reply:
agreed, adding an annotation :)